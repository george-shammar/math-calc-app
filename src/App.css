.App-header {
  font-size: large;
}
