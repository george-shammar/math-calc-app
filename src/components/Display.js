import PropTypes from 'prop-types';
import '../stylesheets/display.css';

const Display = (props) => {
  const { result, next, operation } = props;
  return (
    <div className="bg-display">
      <h1 className="result">{result}</h1>
      <p className="result">{operation}</p>
      <p>{next}</p>
    </div>
  );
};

Display.defaultProps = { result: '0' };
Display.propTypes = {
  result: PropTypes.string,
  next: PropTypes.string.isRequired,
  operation: PropTypes.string.isRequired,
};

export default Display;
