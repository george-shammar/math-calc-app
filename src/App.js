import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <p>Calculator</p>
      </header>
    </div>
  );
}

export default App;
