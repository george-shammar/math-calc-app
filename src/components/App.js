import React from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/Calculate';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      total: null,
      next: null,
      operation: null,
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick = (buttonName) => {
    const data = this.state;
    const calculus = calculate(data, buttonName);
    this.setState({
      total: calculus.total,
      next: calculus.next,
      operation: calculus.operation,
    });
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <>
        <Display result={total} next={next} operation={operation} />

        <ButtonPanel clickHandler={this.handleClick} />
      </>
    );
  }
}

export default App;
