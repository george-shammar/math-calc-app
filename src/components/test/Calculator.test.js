import React from 'react';
import renderer from 'react-test-renderer';
import Display from '../Display';
import App from '../App';

it('renders the Display component correctly', () => {
    const tree = renderer.create(<Display />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders the App component correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders the App component correctly with result props', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});
