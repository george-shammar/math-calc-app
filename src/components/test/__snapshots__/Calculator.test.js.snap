// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the App component correctly 1`] = `
Array [
  <h2>
    Lets do some maths!
  </h2>,
  <div
    className="container"
  >
    <div
      className="bg-display"
    >
      <h1
        className="result"
      />
      <p />
    </div>
    <div>
      <div
        className="button-row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="button-row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
      <div
        className="button-row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="button-row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="button-row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`renders the App component correctly with result props 1`] = `
Array [
  <h2>
    Lets do some maths!
  </h2>,
  <div
    className="container"
  >
    <div
      className="bg-display"
    >
      <h1
        className="result"
      />
      <p />
    </div>
    <div>
      <div
        className="button-row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="button-row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
      <div
        className="button-row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="button-row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="button-row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`renders the Display component correctly 1`] = `
<div
  className="bg-display"
>
  <h1
    className="result"
  >
    0
  </h1>
  <p>
    
  </p>
</div>
`;

exports[`renders the Display component correctly with next props 1`] = `
<div
  className="bg-display"
>
  <h1
    className="result"
  >
    0
  </h1>
  <p>
    
  </p>
</div>
`;

exports[`renders the Display component correctly with operation props 1`] = `
<div
  className="bg-display"
>
  <h1
    className="result"
  >
    0
  </h1>
  <p>
    +
  </p>
</div>
`;

exports[`renders the Display component correctly with result props 1`] = `
<div
  className="bg-display"
>
  <h1
    className="result"
  >
    13
  </h1>
  <p>
    
  </p>
</div>
`;

exports[`renders the Display component correctly with result props 2`] = `
<div
  className="bg-display"
>
  <h1
    className="result"
  >
    20
  </h1>
  <p>
    
  </p>
</div>
`;

exports[`renders the Home component correctly 1`] = `
<div
  className="bg-display"
>
  <h1
    className="result"
  >
    0
  </h1>
  <p>
    
  </p>
</div>
`;
